Libplanet/Action/ActionContext.cs

```diff
@@ -26,7 +26,8 @@ internal class ActionContext : IActionContext
             bool rehearsal = false,
             ITrie? previousBlockStatesTrie = null,
             bool blockAction = false,
-            Predicate<Currency>? nativeTokenPredicate = null)
+            Predicate<Currency>? nativeTokenPredicate = null,
+            System.Action<string>? putLog = null)
```

Comment:
Just curious: is there any rationale that `ActionContext` takes a callback for logging rather than maintaining its own log records?

Reply:
Hmm... I wanted to provide API like `context.PutLog("...")` but it can be achieved by implementing `void PutLog(string log) { Logs.Add(log); }`. I'll try to rewrite the part and request a review again. Thank you!

Reply:
Ah, I remembered why I wrote like it while writing a new code. `IActionContext` is  `IAction.Execute(IActionContext)`'s input. Also, the document says `Contextual data determined by a transaction and a block.`. So I felt strange that the logs are in `IActionContext`. In other words, I felt `IActionContext` should be immutable.

Reply:
But I saw `IActionContext.Random` and `IActionContext.GetUnconsumedContext` and I realized it can be mutable. It was my mistake and I adjusted the part in [9345144](https://github.com/planetarium/libplanet/pull/2505/commits/9345144b3abb89caccd362987bbb4a0ab5d920d9) 😞 🙏🏻